### ISSUE TITLE
Add hgroup tag to list of tags of group block in advanced panel group, html element subgroup

### ISSUE THREAD

**Issue by Lovor01** [WordPress/gutenberg#46781]
## What problem does this address?
`hgroup` is legitimate html tag which is unfortunately seldom used, while it has significant semantic use
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/hgroup
https://html.spec.whatwg.org/multipage/sections.html#the-hgroup-element

## What is your proposed solution?
I suggest `hgroup` tag to be added to the list of tags which you can choose for group block, in order to better model in html the case when title and subtitle(s) are used and when it is appropriate to group them. In this case, `hgroup` is more appropriate than `div` tag.

**Comment by t-hamano** [comment]
I too think this suggestion makes sense.

[W3 dropped `hgroup` in 2013](https://lists.w3.org/Archives/Public/public-html-admin/2013Apr/0003.html). However, it is still alive in the HTML Living Standard and is [no longer deprecated in MDN](https://github.com/mdn/content/pull/20052).

**Comment by carolinan** [comment]
Do we really want this option to be exposed to users, if it is so rarely used?
Developers can already use the `tagName` attribute with the group block to use this tag.

**Comment by t-hamano** [comment]
@carolinan 

> Developers can already use the tagName attribute with the group block to use this tag.

Indeed, since tagName is just a string type, you can also register `hgroup` as a variation, for example:

```javascript
wp.blocks.registerBlockVariation( 'core/group', {
	name: 'hgroup',
	title: 'HGroup',
	attributes: { tagName: 'hgroup' },
} );
```

This approach also available to `query` and `template-part` blocks that can change HTML element.


The key point is whether to allow users to select this as a dropdown, what do you think, @Lovor01?

**Comment by benoitchantre** [comment]
It think that the group block tagName attribute should either be limited to the available options either display the real value if it is not part of the default options (for example hgroup).

**Comment by kraftner** [comment]
I think this should be closed.

`<hgroup>` is not deprecated any more, but now it only allows one `<hX>` with only `<p>`'s before and after, nothing else. This can now also seen in the spec and MDN as linked in the initial issue description.

As such it cannot be a generic container and hence does not (any more) make sense as an option for the group block which does allow any inner content.

If one really wants to use `core/group` with `<hgroup>` it could still be done as variation or pattern, as long as additional logic is somehow implemented to only allow a single `<hx>` and `<p>`'s.

```
<!-- wp:group {"templateLock":"contentOnly"} -->
<div class="wp-block-group"><!-- wp:paragraph -->
<p>supertitle</p>
<!-- /wp:paragraph -->

<!-- wp:heading -->
<h2 class="wp-block-heading">Heading</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>subtitle</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->
```

or

```
<!-- wp:group {"allowedBlocks":["core/paragraph"]} -->
<div class="wp-block-group"><!-- wp:paragraph -->
<p>supertitle</p>
<!-- /wp:paragraph -->

<!-- wp:heading {"lock":{"move":false,"remove":true}} -->
<h2 class="wp-block-heading">Heading</h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>subtitle</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->
```

**Comment by 369work** [comment]
I agree that adding `hgroup` directly to the Group block dropdown could lead to invalid HTML since users might insert unsupported child blocks.

However, I think this highlights a gap: there's currently no native way for users to insert an `hgroup` element from the editor UI without writing block markup manually.

Would a Group block variation with `allowedBlocks` restricted to `core/heading` and `core/paragraph` be a reasonable solution? That way users get access to `hgroup` semantics while the editor enforces the correct content model.

**Comment by kraftner** [comment]
> Would a Group block variation with `allowedBlocks` restricted to `core/heading` and `core/paragraph` be a reasonable solution? That way users get access to `hgroup` semantics while the editor enforces the correct content model.

But what would happen if an existing group gets transformed into that variation? Keep the invalid markup? Drop the inner blocks that aren't valid? I think this makes things even more confusing.

**Comment by Lovor01** [comment]
I agree with @369work. However, because of group block being most generic block of all, it already has variants and perhaps it would be confusing giving hgroup variant offered with row, stack and grid which are layout related variants. I propose adding `hgroup` tag to "HTML ELEMENT" select box along with other semantic html tags, but it would be disabled (or missing, I am not sure if control allows one of elements to be disabled) if group block contained other blocks except heading and paragraph. Also, if `hgroup` would be selected on `core/group` block, it would restrict allowed blocks to `core/heading` and `core/paragraph`, just as @369work wrote.

I guess it would be internally made as group variant, but with special code in `edit.js` to allow functionality as described.

**Comment by Lovor01** [comment]
> > Would a Group block variation with `allowedBlocks` restricted to `core/heading` and `core/paragraph` be a reasonable solution? That way users get access to `hgroup` semantics while the editor enforces the correct content model.
> 
> But what would happen if an existing group gets transformed into that variation? Keep the invalid markup? Drop the inner blocks that aren't valid? I think this makes things even more confusing.

Existing group could not be transformed if inner blocks contained invalid blocks - see my comment above.

**Comment by t-hamano** [comment]
Personally, I suggest closing this issue as "maybe later".

> As such it cannot be a generic container and hence does not (any more) make sense as an option for the group block which does allow any inner content.

That's right. We should avoid adding features that could potentially generate invalid HTML as much as possible. Consumers who genuinely want to provide a correct `hgroup` element can use code like this comment or provide their own block patterns.
